sdk/communication/Azure.Communication.Common/src/CommunicationIdentifier.cs

```diff
@@ -54,25 +70,36 @@ public static CommunicationIdentifier FromRawId(string rawId)
         {
             Argument.AssertNotNullOrEmpty(rawId, nameof(rawId));
 
-            if (rawId.StartsWith("4:", StringComparison.OrdinalIgnoreCase))
+            if (rawId.StartsWith(Phone, StringComparison.OrdinalIgnoreCase))
             {
-                return new PhoneNumberIdentifier(rawId.Substring("4:".Length));
+                return new PhoneNumberIdentifier(rawId.Substring(Phone.Length));
             }
 
             var segments = rawId.Split(':');
             if (segments.Length < 3)
+            {
+                if (segments.Length == 2 && segments[0] == "28")
```

Comment:
I would then use rawId.StartsWith(Bot) to reuse the constants as much as possible.

Reply:
maybe `rawId.StartsWith(Bot, StringComparison.OrdinalIgnoreCase)` will help :)

Reply:
Updated

Reply:
> maybe `rawId.StartsWith(Bot, StringComparison.OrdinalIgnoreCase)` will help :)

@ostoliarova-msft @AikoBB isn't that what I said... :DD

anyways, good that you resolved it 👍